src/main/java/com/jcraft/jsch/ChannelSession.java

```diff
@@ -78,7 +79,7 @@ public void setXForwarding(boolean enable) {
   @Deprecated
   public void setEnv(Hashtable<byte[], byte[]> env) {
     synchronized (this) {
-      this.env = env;
+      this.env = new ConcurrentHashMap<>(env);
```

Comment:
I don't think there is any value in using a ConcurrentHashMap instead of a regular HashMap since all accesses to the env are synchronized on env anyway.